src/orion/algo/space.py

```diff
@@ -825,6 +824,18 @@ def __repr__(self):
         dims = list(self.values())
         return "Space([{}])".format(',\n       '.join(map(str, dims)))
 
+    def items(self):
+        """Return items sorted according to keys"""
+        return [(k, self[k]) for k in self.keys()]
+
+    def values(self):
+        """Return values sorted according to keys"""
+        return [self[k] for k in self.keys()]
+
+    def keys(self):
+        """Return sorted keys"""
+        return list(sorted(super(Space, self).keys()))
```

Comment:
Wouldn't it be more efficient to store an internal list of sorted elements and just insert at the right place when adding new items instead of sorting everytime?

Reply:
Yes but slightly more complicated and then we assume that nothing messes with the ordering during the life of the space. While in comparison this solution ensures the ordering no matter what, for a cost that is most likely unmeasurable considering the small size of spaces. I'd say if this ever turns out to be a bottleneck it should be easy to revisit and improve without affecting the interface.